Jenkinsfile

```diff
@@ -131,38 +178,111 @@ pipeline{
+        string(name: 'ML_RPM', defaultValue: '', description: 'RPM to be used for Image creation. \n If left blank nightly ML rpm will be used.\n Please provide Jenkins accessible path e.g. /project/engineering or /project/qa', trim: true)
+        string(name: 'ML_CONVERTERS', defaultValue: '', description: 'The Converters RPM to be included in the image creation \n If left blank the nightly ML Converters Package will be used.', trim: true)
+        booleanParam(name: 'PUBLISH_IMAGE', defaultValue: false, description: 'Publish image to internal registry')
+    }
+
+    stages {
+        stage('Pre-Build-Check') {
+            steps {
+                PreBuildCheck()
+            }
         }
-        stage("clean") {
-            steps{
-                sh """
-                    cd src/centos
-                    rm -rf *.rpm
-                """
+
+        stage('Copy-RPMs') {
+            steps {
+                CopyRPMs()
+            }
+        }
+
+        stage('Build-Image') {
+            steps {
+                sh "cd src/centos; make build version=${mlVersion}-${env.platformString}-${env.dockerVersion} package=${RPM} converters=${CONVERTERS}"
+            }
+        }
+
+        stage('Image-Test') {
+            steps {
+                RunStructureTests()
+            }
+        }
+
+        stage('Run-Server-Regression-Tests') {
+            steps {
+                RunServerRegressionTests()
+            }
+        }
+
+        stage('Publish-Image') {
+            when {
+                    expression { return params.PUBLISH_IMAGE }
+            }
+            steps {
+                PublishToInternalRegistry()
             }
         }
     }
+
     post {
-        success {  
-            mail bcc: '', body: "<b>Jenkins pipeline for ${env.JOB_NAME} <br>Build Number: ${env.BUILD_NUMBER} <br>${env.BUILD_URL}</b>", cc: '', charset: 'UTF-8', from: '', mimeType: 'text/html', replyTo: '', subject: "BUILD SUCCESS: ${env.JOB_NAME} #${env.BUILD_NUMBER}", to: "${params.passEmail}";
-        }  
-        failure {  
-            mail bcc: '', body: "<b>Jenkins pipeline for ${env.JOB_NAME} <br>Build Number: ${env.BUILD_NUMBER} <br>${env.BUILD_URL}</b>", cc: '', charset: 'UTF-8', from: '', mimeType: 'text/html', replyTo: '', subject: "BUILD ERROR: ${env.JOB_NAME} #${env.BUILD_NUMBER}", to: "${params.failEmail}";
-        }  
-        unstable {  
-            mail bcc: '', body: "<b>Jenkins pipeline for ${env.JOB_NAME} <br>Build Number: ${env.BUILD_NUMBER} <br>${env.BUILD_URL}</b>", cc: '', charset: 'UTF-8', from: '', mimeType: 'text/html', replyTo: '', subject: "BUILD UNSTABLE: ${env.JOB_NAME} #${env.BUILD_NUMBER}", to: "${params.failEmail}";
-        }   
+        always {
+                sh '''
+                    cd src/centos
+                    rm -rf *.rpm
+                '''
+        }
+        success {
+            ResultNotification('BUILD SUCCESS ✅')
```

Comment:
I'm petitioning it to be an animal instead of a checkmark. =P 

🐱  or 🐸  will do.